src/jwt.ts

```diff
@@ -53,37 +58,134 @@ export const decode = (token: string) => {
 };
 
 export const verify = (options: JWTVerifyOptions) => {
+  if (!options.id_token) {
+    throw new Error('ID token is required but missing');
+  }
+
   const decoded = decode(options.id_token);
 
+  if (!decoded.claims.iss) {
+    throw new Error(
+      'Issuer (iss) claim must be a string present in the ID token'
+    );
+  }
+
   if (decoded.claims.iss !== options.iss) {
-    throw new Error('Invalid issuer');
+    throw new Error(
+      `Issuer (iss) claim mismatch in the ID token; expected "${options.iss}", found "${decoded.claims.iss}"`
+    );
   }
-  if (decoded.claims.aud !== options.aud) {
-    throw new Error('Invalid audience');
+
+  if (!decoded.user.sub) {
+    throw new Error(
+      'Subject (sub) claim must be a string present in the ID token'
+    );
   }
+
   if (decoded.header.alg !== 'RS256') {
-    throw new Error('Invalid algorithm');
+    throw new Error(
+      `Signature algorithm of "${decoded.header.alg}" is not supported. Expected the ID token to be signed with "RS256".`
+    );
+  }
+
+  if (
+    !decoded.claims.aud ||
+    !(
+      typeof decoded.claims.aud === 'string' ||
+      Array.isArray(decoded.claims.aud)
+    )
+  ) {
+    throw new Error(
+      'Audience (aud) claim must be a string or array of strings present in the ID token'
+    );
   }
+  if (Array.isArray(decoded.claims.aud)) {
+    if (!decoded.claims.aud.includes(options.aud)) {
+      throw new Error(
+        `Audience (aud) claim mismatch in the ID token; expected "${
+          options.aud
+        }" but was not one of "${decoded.claims.aud.join(', ')}"`
+      );
+    }
+    if (decoded.claims.aud.length > 1) {
+      if (!decoded.claims.azp) {
+        throw new Error(
+          'Authorized Party (azp) claim must be a string present in the ID token when Audience (aud) claim has multiple values'
+        );
+      }
+      if (decoded.claims.azp !== options.client_id) {
+        throw new Error(
+          `Authorized Party (azp) claim mismatch in the ID token; expected "${options.aud}", found "${decoded.claims.azp}"`
+        );
+      }
+    }
+  } else if (decoded.claims.aud !== options.aud) {
+    throw new Error(
+      `Audience (aud) claim mismatch in the ID token; expected "${options.aud}" but found "${decoded.claims.aud}"`
+    );
+  }
+
+  if (!decoded.claims.nonce) {
```

Comment:
should be optional unless your sdk ensures a value is always sent, even when user sets "null/undefined"

```suggestion
  if (options.nonce && !decoded.claims.nonce) {
```

Reply:
fixed